### PR TITLE
[`v3`] Use `torch.arange` instead of `torch.tensor(range(...))`

### PR DESCRIPTION
**Pull request by tomaarsen**
See https://github.com/UKPLab/sentence-transformers/pull/2449#issuecomment-2114787400

Hello!

## Pull Request overview
* Don't override the labels variable to avoid inplace operation

## Details
I'm not sure if this is the cause of the issue as it doesn't really seem like an inplace operation; but I think it might be, and it's one of the only places with a LongTensor.

@Jakobhenningjensen would you be able to rerun your experiment with this branch instead?
i.e.
```
pip install git+https://github.com/tomaarsen/sentence-transformers.git@v3/mnrl_avoid_inplace_op
```
or 
```
git clone https://github.com/tomaarsen/sentence-transformers
cd sentence-transformers
git checkout v3/mnrl_avoid_inplace_op
```
or related commands.

- Tom Aarsen

### CODE REVIEW

**Comment by Jakobhenningjensen** [comment]
It seems like the problem still persists

**Comment by tomaarsen** [comment]
Damn. I'll need to read up more on inplace operations then.

**Comment by tomaarsen** [comment]
Could you try
```python
model = SentenceTransformer("intfloat/multilingual-e5-small", device="cuda")
del model[2]
...
```
This removes the Normalize layer, which could be the culprit.

- Tom Aarsen

**Comment by Jakobhenningjensen** [comment]
Surely - on which branch? Still this or the "ordinary v3" ?

**Comment by tomaarsen** [comment]
Either one is fine

**Comment by tomaarsen** [comment]
I think I'm able to reproduce this now, will investigate to see what's causing the issue.

- Tom Aarsen

**Comment by Jakobhenningjensen** [comment]
> I think I'm able to reproduce this now, will investigate to see what's causing the issue.
> 
> - Tom Aarsen

Great! I'm currently trying the `del model[2]` approach but our deployment pipeline is having some issues

**Comment by Jakobhenningjensen** [comment]
Unfortunately the error still persists

**Comment by tomaarsen** [comment]
Merging this, as I prefer this `arange` over the old approach, but I'll put the inplace fix in another PR.
